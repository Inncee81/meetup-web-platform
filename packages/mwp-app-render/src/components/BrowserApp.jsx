import React from 'react';
import PropTypes from 'prop-types';
import BrowserRouter from 'react-router-dom/BrowserRouter';
import PlatformApp from './shared/PlatformApp';
import initClickTracking from 'mwp-tracking-plugin/lib/util/browserInit';
import { BrowserCookieProvider } from '@meetup/mwp-cookie/lib/BrowserCookieProvider';

/**
 * A simple component to wrap the base PlatformApp with the BrowserRouter
 */
class BrowserApp extends React.Component {
	componentDidMount() {
		initClickTracking();
	}
	render() {
		const { appContext, store, routes } = this.props;
		return (
			<BrowserRouter basename={appContext.basename}>
				<BrowserCookieProvider>
					<PlatformApp
						appContext={appContext}
						store={store}
						routes={routes}
					/>
				</BrowserCookieProvider>
			</BrowserRouter>
		);
	}
}

BrowserApp.propTypes = {
	routes: PropTypes.array.isRequired,
	store: PropTypes.object.isRequired,
	appContext: PropTypes.object.isRequired,
};
BrowserApp.defaultProps = {
	basename: '',
};

export default BrowserApp;
