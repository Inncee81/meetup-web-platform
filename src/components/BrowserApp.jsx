import React from 'react';
import PropTypes from 'prop-types';
import { getBrowserCreateStore } from '../util/createStoreBrowser';
import BrowserRouter from 'react-router-dom/BrowserRouter';
import PlatformApp from '../components/PlatformApp';

/**
 * This function creates a 'renderer', which is just a function that, when
 * called, will call ReactDOM.render() to render the application
 *
 * The routes, reducer, and app-specific middleware are provided by the
 * application - everything else is general to the meetup web platform
 *
 * @param {Object} routes the React Router routes object
 * @param {Function} reducer the root Redux reducer for the app
 * @param {Function} middleware (optional) any app-specific middleware that
 *   should be applied to the store
 * @param {String} baseUrl an optional baseUrl to serve the site from,
 * 	_including_ surrounding slashes, e.g. '/en-US/'
 *
 * @returns {Function} a function that results in a ReactDOM.render call - can
 *   use a custom root element ID or default to `'outlet'`
 */
class BrowserApp extends React.Component {
	constructor(props) {
		super(props);
		// the initial state is delivered in the HTML from the server as a plain object
		// containing the HTML-escaped JSON string in `window.INITIAL_STATE.escapedState`.
		// unescape the text using native `textarea.textContent` unescaping
		const escape = document.createElement('textarea');
		escape.innerHTML = window.APP_RUNTIME.escapedState;
		const unescapedStateJSON = escape.textContent;
		const initialState = JSON.parse(unescapedStateJSON);
		const createStore = getBrowserCreateStore(props.routes, props.middleware, props.baseUrl);
		this.store = createStore(props.reducer, initialState);
	}
	render() {
		const { baseUrl, routes } = this.props;
		return (
			<BrowserRouter basename={baseUrl}>
				<PlatformApp store={this.store} routes={routes} />
			</BrowserRouter>
		);
	}
}

BrowserApp.propTypes = {
	routes: PropTypes.array.isRequired,
	reducer: PropTypes.func.isRequired,
	middleware: PropTypes.array.isRequired,
	baseUrl: PropTypes.string.isRequired,
};
BrowserApp.defaultProps = {
	middleware: [],
	baseUrl: '',
};

export default BrowserApp;

