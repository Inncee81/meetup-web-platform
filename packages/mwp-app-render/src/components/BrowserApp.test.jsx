import React from 'react';
import { shallow } from 'enzyme';
import BrowserApp from './BrowserApp';

jest.mock('./ApolloProvider', () => {
	return props => <div {...props} />;
});

// TODO: skipping test until browser test environment enabled
describe.skip('BrowserApp', function() {
	const routes = [];
	const basename = '';
	const appWrapper = shallow(
		<BrowserApp routes={routes} store={{}} basename={basename} />
	);
	it('exists', function() {
		expect(appWrapper).toMatchSnapshot();
	});
});
