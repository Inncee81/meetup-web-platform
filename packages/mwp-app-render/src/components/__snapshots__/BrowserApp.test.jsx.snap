// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserApp exists 1`] = `
<BrowserRouter
  basename=""
>
  <PlatformApp
    routes={Array []}
    store={Object {}}
  />
</BrowserRouter>
`;
